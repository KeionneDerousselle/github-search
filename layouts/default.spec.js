import DefaultLayout from './default'

describe('Default Layout', () => {
  let wrapper

  beforeAll(() => {
    wrapper = mountPreMocked(DefaultLayout)
  })

  afterAll(() => {
    wrapper.destroy()
  })

  it('should render as expected', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the header', () => {
    expect(wrapper.get('header#kd-github-search-header').element).toBeVisible()
  })

  it('should render the content', () => {
    expect(wrapper.get('main#kd-github-search-content').element).toBeVisible()
  })

  it('should render the footer', () => {
    expect(wrapper.get('footer#kd-github-search-footer').element).toBeVisible()
  })
})
