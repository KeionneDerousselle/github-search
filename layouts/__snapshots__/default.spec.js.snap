// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Layout should render as expected 1`] = `
<div id="kd-github-search-app" class="github-search-app">
  <div class="github-search-container">
    <header id="kd-github-search-header" class="header github-search-header"></header>
    <main id="kd-github-search-content" class="github-search-content">
      <nuxt-stub></nuxt-stub>
    </main>
    <footer id="kd-github-search-footer" class="footer github-search-footer">
      <div class="footer__content"><span>
      Created with
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" fill="currentColor" class="w-4 inline-flex fill-current text-pink-900"><path fill-rule="evenodd" d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z" clip-rule="evenodd"></path></svg>
      by <a id="footer-dev-link" href="https://www.keionne.com" target="_blank" class="footer__link">
        Keionne Derousselle<span class="sr-only">Opens the developer's website in a new window.</span>.
        </a></span> <span>
      Take a look at the code
      <a id="footer-repo-link" href="https://www.github.com/KeionneDerousselle/github-search" target="_blank" class="footer__link">
        here<span class="sr-only">Opens the source code repository for this website in a new window.</span>.
        </a></span></div>
    </footer>
  </div>
</div>
`;
